GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -423,6 +441,25 @@ private void SetRowCount(int count)
                 {
                     RowCount = count;
                 }
+
+                if (ToBeSelectedRowIndexes.Any() &&
+                    ToBeSelectedRowIndexes.TryPeek(out int toBeSelectedRowIndex) &&
+                    count > toBeSelectedRowIndex)
+                {
+                    try
+                    {
+                        ToBeSelectedRowIndexes.TryDequeue(out int ignore);
```

Comment:
I think you can write `out _` to ignore the value.

Also this is a concurrent collection but this code races. Is that ok?

Reply:
Fixed directly in master